### DISCUSSION TITLE
How to get last state

### DISCUSSION THREAD

**Discussion by BeanWei** [ueberdosis/tiptap#1961] [General Questions]
Is it possible to get last state (last input node type).

I am working on my react rich-text editor, but the menubar re-render when editing. So i need the last editing status to control it. I had ignore some menubar re-render, its needn't `isActive`  function. 

Sorry! My english is pool, can anyone help me?

**Comment by omarkhairy21** [comment]
if you use react use `useMemo` to prevent unnecessarily rerender
